internal/tenant/tenanttest/tenant_test.go

```diff
@@ -121,3 +128,22 @@ func newTestCallResourceResponseSender() *testCallResourceResponseSender {
 func (s *testCallResourceResponseSender) Send(_ *backend.CallResourceResponse) error {
 	return nil
 }
+
+// getFreePort returns a random free port listening on 127.0.0.1.
+func getFreePort() (port int, err error) {
```

Comment:
Although this is a utility, this breaks my brain. I think it's because we have a named `port` but it's not referenced. I assume the main reason for this pattern here is because of returning the error from the `defer`?

Reply:
Yeah it's because of that, however `defer` is not required in that case. I have remoed that :smile: